Assert error on inline fail: "assertion failed: method *: in trait Tuple"

## Description

I'm in the process of migrating some internal services to Scala 3.

In code, that should generate some config readers, the compiler crashes.
The `given instance exportReader is declared as `inline`, but was not inlined` error can not be fixed by increasing `-Xmax-inlines`,
but the actual crash comes after: `java.lang.AssertionError: assertion failed: method *: in trait Tuple`

## Compiler version

3.3.4 and 3.6.3

## Minimized code

```Scala
//> using dep com.github.pureconfig::pureconfig-core:0.17.8
//> using dep com.github.pureconfig::pureconfig-generic-scala3:0.17.8

import pureconfig.*
import pureconfig.generic.*
import pureconfig.generic.semiauto.*
import scala.deriving.Mirror

inline given exportReader[A: Mirror.Of]: Exported[ConfigReader[A]] = Exported(deriveReader[A])

val config = ConfigSource.string("{ test: { matchers: [ { rules: [ { something: \"foo\" } ] }] } }")

case class Rule(something: Option[String])
case class Matchers(rules: List[Rule])
case class Settings(matchers: List[Matchers])

class Consumer(settings: Settings)

val c: Consumer = {
  val settings = config.at("test").loadOrThrow[Settings]
  new Consumer(settings)
}
```

## Output (click arrow to expand)
<!--
Note: to get a full crash log when using scala-cli, run compilation with the flag `--server=false`
-->
<details>

```scala
-- Error: /home/ben/.scala-build/ben_ef31ba4fc4-6820637c46/src_generated/main/repro.scala:26:56 
26 |  val settings = config.at("test").loadOrThrow[Settings]
   |                                                        ^
   |given instance exportReader is declared as `inline`, but was not inlined
   |
   |Try increasing `-Xmax-inlines` above 32
   |----------------------------------------------------------------------------
   |Inline stack trace
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareConfigReaderDerivation.scala:21
15 |inline given exportReader[A: Mirror.Of]: Exported[ConfigReader[A]] = Exported(deriveReader[A])
   |                                                                              ^^^^^^^^^^^^^^^
    ----------------------------------------------------------------------------
-- Error: /home/ben/.scala-build/ben_ef31ba4fc4-6820637c46/src_generated/main/repro.scala:26:56 
26 |  val settings = config.at("test").loadOrThrow[Settings]
   |                                                        ^
   |       method readCaseClass is declared as `inline`, but was not inlined
   |
   |       Try increasing `-Xmax-inlines` above 32
   |----------------------------------------------------------------------------
   |Inline stack trace
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:79
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:79
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:79
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:79
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:79
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:79
15 |inline given exportReader[A: Mirror.Of]: Exported[ConfigReader[A]] = Exported(deriveReader[A])
   |                                                                              ^^^^^^^^^^^^^^^
    ----------------------------------------------------------------------------
-- Error: /home/ben/.scala-build/ben_ef31ba4fc4-6820637c46/src_generated/main/repro.scala:26:56 
26 |  val settings = config.at("test").loadOrThrow[Settings]
   |                                                        ^
   |               method widen is declared as `inline`, but was not inlined
   |
   |               Try increasing `-Xmax-inlines` above 32
   |----------------------------------------------------------------------------
   |Inline stack trace
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:81
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:81
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:81
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:81
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:81
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   |This location contains code that was inlined from HintsAwareProductConfigReaderDerivation.scala:81
15 |inline given exportReader[A: Mirror.Of]: Exported[ConfigReader[A]] = Exported(deriveReader[A])
   |                                                                              ^^^^^^^^^^^^^^^
    ----------------------------------------------------------------------------

  unhandled exception while running erasure on /home/ben/.scala-build/ben_ef31ba4fc4-6820637c46/src_generated/main/repro.scala

  An unhandled exception was thrown in the compiler.
  Please file a crash report here:
  https://github.com/scala/scala3/issues/new/choose
  For non-enriched exceptions, compile with -Xno-enrich-error-messages.

     while compiling: /home/ben/.scala-build/ben_ef31ba4fc4-6820637c46/src_generated/main/repro.scala
        during phase: erasure
                mode: Mode(ImplicitsEnabled)
     library version: version 2.13.14
    compiler version: version 3.5.2
            settings: -classpath /home/ben/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.5.2/scala3-library_3-3.5.2.jar:/home/ben/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/pureconfig/pureconfig-core_3/0.17.8/pureconfig-core_3-0.17.8.jar:/home/ben/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/pureconfig/pureconfig-generic-scala3_3/0.17.8/pureconfig-generic-scala3_3-0.17.8.jar:/home/ben/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.14/scala-library-2.13.14.jar:/home/ben/.cache/coursier/v1/https/repo1.maven.org/maven2/com/typesafe/config/1.4.3/config-1.4.3.jar:/home/ben/.cache/coursier/v1/https/repo1.maven.org/maven2/com/github/pureconfig/pureconfig-generic-base_3/0.17.8/pureconfig-generic-base_3-0.17.8.jar -d /home/ben/.scala-build/ben_ef31ba4fc4-6820637c46/classes/main -sourceroot /home/ben

Exception in thread "main" java.lang.AssertionError: assertion failed: method *: in trait Tuple
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
	at dotty.tools.dotc.transform.Erasure$Typer.mapOwner$1(Erasure.scala:697)
	at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:709)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3390)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:823)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3435)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:850)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:863)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:863)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.$anonfun$64(Typer.scala:2834)
	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:256)
	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2834)
	at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:969)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3397)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1399)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1403)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:863)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:863)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2771)
	at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:920)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3394)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1399)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1403)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:640)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3420)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
	at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:911)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3438)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:678)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3390)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:823)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3435)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:850)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedReturn(Typer.scala:2248)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3431)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1406)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.$anonfun$16(Typer.scala:1500)
	at dotty.tools.dotc.typer.Applications.harmonic(Applications.scala:2559)
	at dotty.tools.dotc.typer.Applications.harmonic$(Applications.scala:434)
	at dotty.tools.dotc.typer.Typer.harmonic(Typer.scala:145)
	at dotty.tools.dotc.typer.Typer.typedIf(Typer.scala:1503)
	at dotty.tools.dotc.transform.Erasure$Typer.typedIf(Erasure.scala:896)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3424)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3630)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1399)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1403)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedLabeled(Typer.scala:2198)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3408)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.$anonfun$64(Typer.scala:2834)
	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:256)
	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2834)
	at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:969)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3397)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1399)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1403)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:863)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:863)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1406)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.$anonfun$64(Typer.scala:2834)
	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:256)
	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2834)
	at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:969)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3397)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:3097)
	at dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1058)
	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3403)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3407)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1399)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1403)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:640)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3420)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3574)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
	at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:911)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3438)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1406)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:640)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3420)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
	at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:911)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3438)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1406)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:640)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3420)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3574)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
	at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:911)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3438)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3574)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
	at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:911)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3438)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:863)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:863)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:640)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3420)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
	at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:911)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3438)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:863)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:863)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:863)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:161)
	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:863)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3415)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2771)
	at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:920)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3394)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1399)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1403)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3423)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.typer.Typer.$anonfun$64(Typer.scala:2834)
	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:256)
	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2834)
	at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:969)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3397)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:3097)
	at dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1058)
	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3403)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3407)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3499)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3603)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3649)
	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1082)
	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:3230)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3449)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3500)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3577)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3581)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3692)
	at dotty.tools.dotc.transform.Erasure.run(Erasure.scala:145)
	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:380)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.immutable.List.foreach(List.scala:334)
	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:373)
	at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:343)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at dotty.tools.dotc.Run.runPhases$1(Run.scala:336)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:384)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:396)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:396)
	at dotty.tools.dotc.Run.compileSources(Run.scala:282)
	at dotty.tools.dotc.Run.compile(Run.scala:267)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
	at dotty.tools.dotc.Driver.process(Driver.scala:201)
	at dotty.tools.dotc.Driver.process(Driver.scala:169)
	at dotty.tools.dotc.Driver.process(Driver.scala:181)
	at dotty.tools.dotc.Driver.main(Driver.scala:211)
	at dotty.tools.dotc.Main.main(Main.scala)
Compilation failed
```
</details>

Minimised:
```scala
import scala.deriving.Mirror
import scala.compiletime._
import scala.compiletime.ops.int._

object HintsAwareConfigReaderDerivation {
  inline def deriveReader[A]: ConfigReader[A] =
    summonFrom { case given Mirror.Of[A] =>
      deriveReaderWithMirror[A]
    }

  private inline def summonConfigReader[A]: ConfigReader[A] =
    summonFrom { case reader: ConfigReader[A] => reader }

  private inline def deriveReaderWithMirror[A](using m: Mirror.Of[A]): ConfigReader[A] =
    inline m match {
      case pm: Mirror.ProductOf[A] => deriveProductReader[A](using pm)
      case sm: Mirror.SumOf[A] => deriveSumReader[A](using sm)
    }

  inline def deriveSumReader[A](using cm: Mirror.SumOf[A]): ConfigReader[A] =
    summonAllConfigReaders[cm.MirroredElemTypes, A]
    ???

  private inline def summonAllConfigReaders[T <: Tuple, A]: List[ConfigReader[A]] =
    inline erasedValue[T] match {
      case _: (h *: t) =>
        (summonConfigReader[h] :: summonAllConfigReaders[t, A]).asInstanceOf[List[ConfigReader[A]]]
      case _: EmptyTuple => Nil
    }
  
  inline def deriveProductReader[A](using pm: Mirror.ProductOf[A]): ConfigReader[A] =
    val tupleSize = summonInline[ValueOf[Tuple.Size[pm.MirroredElemTypes]]]
    val result = readCaseClass[pm.MirroredElemTypes, 0, A]().getOrElse(???)
    pm.fromProduct(result)
    ???

  private inline def readCaseClass[T <: Tuple, N <: Int, A](): Either[List[String], T] =
    inline erasedValue[T] match {
      case _: (h *: t) =>
        lazy val reader = summonConfigReader[h]
        val head: Either[List[String], A] = ???
        val tail = readCaseClass[t, N + 1, A]()
        val resultTuple = ConfigReader.Result.zipWith(head, tail)((h, t) => h *: t) // causes crash
        ???

      case _: EmptyTuple => ???
    }
}

trait ConfigReader[A]
object ConfigReader {
  implicit def optionReader[A](implicit conv: ConfigReader[A]): ConfigReader[Option[A]] = ???
  implicit def traversableReader[A, F[A] <: TraversableOnce[A]](implicit configConvert: ConfigReader[A]): ConfigReader[F[A]] = ???
  // implicit def stringReader: ConfigReader[String] = ??? // uncommenting fixes this
  implicit def exportedReader[A](implicit exported: Exported[ConfigReader[A]]): ConfigReader[A] = exported.instance
  case class Exported[A](instance: A)
  object Result {
    def zipWith[A, B, C](first: Either[List[String], A], second: Either[List[String], B])(
        f: (A, B) => C
    ): Either[List[String], C] = ???
  }
}

import ConfigReader._
inline given exportReader[A: Mirror.Of]: Exported[ConfigReader[A]] = Exported(HintsAwareConfigReaderDerivation.deriveReader[A])

case class Rule(something: Option[String])
case class Settings(rules: List[Rule])

val settings =
  summon[ConfigReader[Settings]]

```

<details>

```scala
[error] ./repr.scala:72:33
[error] given instance exportReader is declared as `inline`, but was not inlined
[error] 
[error] Try increasing `-Xmax-inlines` above 32
[error]   summon[ConfigReader[Settings]]
[error]                                 ^
Error compiling project (Scala 3.3.5, JVM (17))
Error: java.lang.AssertionError: assertion failed: method *: in trait Tuple
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at dotty.tools.dotc.transform.Erasure$Typer.mapOwner$1(Erasure.scala:689)
        at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:701)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3095)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:817)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3145)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:844)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.$anonfun$58(Typer.scala:2563)
        at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:242)
        at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2563)
        at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:963)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3102)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:670)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3095)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:817)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3145)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:844)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.$anonfun$58(Typer.scala:2563)
        at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:242)
        at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2563)
        at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:963)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3102)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:670)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3095)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:817)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3145)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:844)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:824)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3145)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3330)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.$anonfun$58(Typer.scala:2563)
        at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:242)
        at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2563)
        at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:963)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3102)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:670)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3095)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:817)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3145)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:844)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1216)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1219)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3134)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:632)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3131)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:100)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
        at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:155)
        at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3126)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2501)
        at dotty.tools.dotc.transform.Erasure$Typer.typedValDef(Erasure.scala:914)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3099)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1212)
        at dotty.tools.dotc.typer.ReTyper.typedInlined(ReTyper.scala:99)
        at dotty.tools.dotc.transform.Erasure$Typer.typedInlined(Erasure.scala:905)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3148)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.typer.Typer.$anonfun$58(Typer.scala:2563)
        at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:242)
        at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2563)
        at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:963)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3102)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2746)
        at dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1051)
        at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3114)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3118)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3206)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3303)
        at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3349)
        at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1062)
        at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2889)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3159)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3207)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3277)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3281)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3392)
        at dotty.tools.dotc.transform.Erasure.run(Erasure.scala:145)
        at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:337)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.immutable.List.foreach(List.scala:334)
        at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:333)
        at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:315)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
        at dotty.tools.dotc.Run.runPhases$1(Run.scala:308)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:349)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:358)
        at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
        at dotty.tools.dotc.Run.compileUnits(Run.scala:358)
        at dotty.tools.dotc.Run.compileSources(Run.scala:261)
        at dotty.tools.dotc.Run.compile(Run.scala:246)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
        at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:141)
        at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
        at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
        at sbt.internal.inc.bloop.internal.BloopHighLevelCompiler.compileSources$1(BloopHighLevelCompiler.scala:148)
        at sbt.internal.inc.bloop.internal.BloopHighLevelCompiler.$anonfun$compile$9(BloopHighLevelCompiler.scala:181)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at sbt.internal.inc.bloop.internal.BloopHighLevelCompiler.$anonfun$compile$1(BloopHighLevelCompiler.scala:73)
        at bloop.tracing.NoopTracer$.trace(BraveTracer.scala:53)
        at sbt.internal.inc.bloop.internal.BloopHighLevelCompiler.timed$1(BloopHighLevelCompiler.scala:72)
        at sbt.internal.inc.bloop.internal.BloopHighLevelCompiler.$anonfun$compile$8(BloopHighLevelCompiler.scala:181)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at monix.eval.internal.TaskRunLoop$.startFull(TaskRunLoop.scala:81)
        at monix.eval.internal.TaskRestartCallback.syncOnSuccess(TaskRestartCallback.scala:101)
        at monix.eval.internal.TaskRestartCallback.onSuccess(TaskRestartCallback.scala:74)
        at monix.eval.internal.TaskExecuteOn$AsyncRegister$$anon$1.run(TaskExecuteOn.scala:71)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```
</details>

It looks to me like somehow due to the way inlining is handled in pureconfig, by the time we are looking for ConfigReader[String], it's no longer in implicit scope, leading to it trying to construct ConfigReader[String], leading to some kind of inifinite recursion, where we always end up with the innermost exportReader reaching the limit. I need to investigate further. Either way, the crash should not happen, and is caused by an unrelated element.

--- a/compiler/src/dotty/tools/dotc/typer/Implicits.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Implicits.scala
@@ -1294,11 +1294,15 @@ trait Implicits:
         val history = ctx.searchHistory.nest(cand, pt)
         val typingCtx =
           searchContext().setNewTyperState().setFreshGADTBounds.setSearchHistory(history)
+        val alreadyStoppedInlining = ctx.base.stopInlining
         val result = typedImplicit(cand, pt, argument, span)(using typingCtx)
         result match
           case res: SearchSuccess =>
             ctx.searchHistory.defineBynameImplicit(wideProto, res)
           case _ =>
+            if !alreadyStoppedInlining && ctx.base.stopInlining then
+              // a call overflowed as part of the expansion when typing the implicit
+              ctx.base.stopInlining = false
             // Since the search failed, the local typerstate will be discarded
             // without being committed, but type variables local to that state
             // might still appear in an error message, so we run `gc()` here to

--- a/tests/neg/i13044.check
+++ b/tests/neg/i13044.check
@@ -1,63 +1,18 @@
--- Error: tests/neg/i13044.scala:61:40 ---------------------------------------------------------------------------------
-61 |   implicit def typeSchema: Schema[A] = Schema.gen // error // error
+-- [E172] Type Error: tests/neg/i13044.scala:61:40 ---------------------------------------------------------------------
+61 |   implicit def typeSchema: Schema[A] = Schema.gen // error
    |                                        ^^^^^^^^^^
-   |                                        given instance gen is declared as `inline`, but was not inlined
+   |                                 No given instance of type Schema[B] was found.
+   |                                 I found:
    |
-   |                                        Try increasing `-Xmax-inlines` above 32
+   |                                     Schema.gen[B]
+   |
+   |                                 But given instance gen in trait SchemaDerivation does not match type Schema[B].
    |--------------------------------------------------------------------------------------------------------------------
    |Inline stack trace
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from i13044.scala:17
 17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:17
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
+   |                      ^^^^^^^^^^^^^^^^^^^^^^^
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from i13044.scala:17
 18 |        builder :: recurse[ts]
@@ -68,79 +23,6 @@
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from i13044.scala:17
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-    --------------------------------------------------------------------------------------------------------------------
--- Error: tests/neg/i13044.scala:61:40 ---------------------------------------------------------------------------------
-61 |   implicit def typeSchema: Schema[A] = Schema.gen // error // error
-   |                                        ^^^^^^^^^^
-   |                                        method recurse is declared as `inline`, but was not inlined
-   |
-   |                                        Try increasing `-Xmax-inlines` above 32
-   |--------------------------------------------------------------------------------------------------------------------
-   |Inline stack trace
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-18 |        builder :: recurse[ts]
-   |                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-33 |  inline given gen[A]: Schema[A] = derived
-   |                                   ^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-17 |        val builder = summonInline[Schema[t]].asInstanceOf[Schema[Any]]
-   |                                   ^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-18 |        builder :: recurse[ts]
-   |                   ^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
-29 |        lazy val fields = recurse[m.MirroredElemTypes]
-   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-   |This location contains code that was inlined from i13044.scala:18
 33 |  inline given gen[A]: Schema[A] = derived
    |                                   ^^^^^^^
     --------------------------------------------------------------------------------------------------------------------

--- a/tests/neg/i13044.scala
+++ b/tests/neg/i13044.scala
@@ -58,5 +58,5 @@ case class B(c: C)
 case class A(a: A, b: B)
 
 object TestApp {
-   implicit def typeSchema: Schema[A] = Schema.gen // error // error
+   implicit def typeSchema: Schema[A] = Schema.gen // error
 }

--- a/tests/neg/i22423.check
+++ b/tests/neg/i22423.check
@@ -1,0 +1,24 @@
+-- Error: tests/neg/i22423.scala:35:14 ---------------------------------------------------------------------------------
+35 |  exportReader[Settings] // error
+   |  ^^^^^^^^^^^^^^^^^^^^^^
+   |  cannot reduce summonFrom with
+   |   patterns :  case given reader @ _:ConfigReader[List[String]]
+   |--------------------------------------------------------------------------------------------------------------------
+   |Inline stack trace
+   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   |This location contains code that was inlined from i22423.scala:12
+12 |    summonFrom { case reader: ConfigReader[A] => reader }
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   |This location contains code that was inlined from i22423.scala:12
+15 |    summonConfigReader[List[String]]
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   |This location contains code that was inlined from i22423.scala:12
+ 8 |    readCaseClass()
+   |    ^^^^^^^^^^^^^^^
+   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   |This location contains code that was inlined from i22423.scala:12
+30 |inline given exportReader[A]: Exported[ConfigReader[A]] = Exported(HintsAwareConfigReaderDerivation.deriveReader[A])
+   |                                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    --------------------------------------------------------------------------------------------------------------------

--- a/tests/neg/i22423.scala
+++ b/tests/neg/i22423.scala
@@ -1,0 +1,35 @@
+//> using options -Xmax-inlines:7
+import scala.deriving.Mirror
+import scala.compiletime._
+import scala.compiletime.ops.int._
+
+object HintsAwareConfigReaderDerivation {
+  inline def deriveReader[A]: ConfigReader[A] =
+    readCaseClass()
+    ???
+
+  private inline def summonConfigReader[A]: ConfigReader[A] =
+    summonFrom { case reader: ConfigReader[A] => reader }
+
+  private inline def readCaseClass(): Unit =
+    summonConfigReader[List[String]]
+    val a1: Int = ???
+    val a2: EmptyTuple = ???
+    a1 *: a2
+    ???
+}
+
+trait ConfigReader[A]
+object ConfigReader {
+  implicit def traversableReader[A, F[A] <: TraversableOnce[A]](implicit configConvert: ConfigReader[A]): ConfigReader[F[A]] = ???
+  implicit def exportedReader[A](implicit exported: Exported[ConfigReader[A]]): ConfigReader[A] = exported.instance
+  case class Exported[A](instance: A)
+}
+
+import ConfigReader._
+inline given exportReader[A]: Exported[ConfigReader[A]] = Exported(HintsAwareConfigReaderDerivation.deriveReader[A])
+
+case class Settings(rules: List[String])
+
+val settings =
+  exportReader[Settings] // error

